services/core-api/app/api/verifiable_credentials/manager.py

```diff
@@ -43,81 +46,130 @@ def revoke_all_credentials_for_permit(permit_guid: str, mine_guid:str, reason:st
         credential_attrs["mine_no"] = permit_amendment.mine.mine_no
-        credential_attrs["issue_date"] = int(permit_amendment.issue_date.strftime("%Y%m%d")) if is_feature_enabled(Feature.VC_MINES_ACT_PERMIT_20) else permit_amendment.issue_date
+        credential_attrs["issue_date"] = int(
+            permit_amendment.issue_date.strftime("%Y%m%d")) if is_feature_enabled(
+                Feature.VC_MINES_ACT_PERMIT_20) else permit_amendment.issue_date
         # https://github.com/hyperledger/aries-rfcs/tree/main/concepts/0441-present-proof-best-practices#dates-and-predicates
         credential_attrs["latitude"] = permit_amendment.mine.latitude
         credential_attrs["longitude"] = permit_amendment.mine.longitude
         credential_attrs["bond_total"] = permit_amendment.permit.active_bond_total
-        credential_attrs["tsf_operating_count"] = len([tsf for tsf in permit_amendment.mine.mine_tailings_storage_facilities if tsf.tsf_operating_status_code == "OPT"])
-        credential_attrs["tsf_care_and_maintenance_count"] = len([tsf for tsf in permit_amendment.mine.mine_tailings_storage_facilities if tsf.tsf_operating_status_code == "CAM"])
+        credential_attrs["tsf_operating_count"] = len([
+            tsf for tsf in permit_amendment.mine.mine_tailings_storage_facilities
+            if tsf.tsf_operating_status_code == "OPT"
+        ])
+        credential_attrs["tsf_care_and_maintenance_count"] = len([
+            tsf for tsf in permit_amendment.mine.mine_tailings_storage_facilities
+            if tsf.tsf_operating_status_code == "CAM"
+        ])
 
         # offer credential
-        attributes = [{
-            # "mime-type":"text/plain",
-            # NB Orbit does not expect this removing for now
-            "name":str(attr),
-            "value":str(val),
-        } for attr,val in credential_attrs.items()]
+        attributes = [
+            {
+                                                        # "mime-type":"text/plain",
+                                                        # NB Orbit does not expect this removing for now
+                "name": str(attr),
+                "value": str(val),
+            } for attr, val in credential_attrs.items()
+        ]
 
         return attributes
-    
+
+    @classmethod
+    def revoke_all_credentials_for_permit(cls, permit_guid: str):
+        pass
+
     @classmethod
-    def revoke_all_credentials_for_permit(permit_guid: str):
-        pass
\ No newline at end of file
+    def produce_map_01_credential_payload(cls, did: str, permit_amendment: PermitAmendment):
+
+        credential = {
+            "@context":
+            ["https://www.w3.org/2018/credentials/v1", {
+                "@vocab": "urn:bcgov:attributes#"
+            }],
+            "type": ["VerifiableCredential"],
+            "issuer": {
+                "id": did,
+            },
+            "issuanceDate": "2010-01-01T19:23:24Z",
```

Comment:
Just curious if the issuanceDate is supposed to be hardcoded?

Reply:
It is not. it should be the issue_date (and should actually be removed from the contents).